crates/amaru-ledger/src/rules/transaction/inputs.rs

```diff
@@ -90,7 +92,7 @@ where
                     };
                 };
             }
-            None => unimplemented!("failed to lookup input: {input:?}"),
+            None => Err(InvalidInputs::UnknownInput)?,
```

Comment:
The error lacks a bit of context, we should at least provides the input's reference.

Reply:
👍